autogpt/commands/web_requests.py

```diff
@@ -4,20 +4,54 @@
 from urllib.parse import urljoin, urlparse
 
 import requests
+import traceback
 from bs4 import BeautifulSoup
 from requests import Response
 from requests.compat import urljoin
+from typing import Dict, Any, Union, List
 
 from autogpt.config import Config
 from autogpt.memory import get_memory
 from autogpt.processing.html import extract_hyperlinks, format_hyperlinks
+from autogpt.commands.command import command
 
 CFG = Config()
 memory = get_memory(CFG)
 
 session = requests.Session()
 session.headers.update({"User-Agent": CFG.user_agent})
 
+JSONType = Union[Dict[str, Any], List[Any], str, int, float, bool, None]
+
+@command(
+    "make_http_request",
+    "Make an HTTP request",
+    '"url": "<url>", "method": "<method>", "auth_token": "<auth_token>", "data": "<data>"',
+)
+def make_http_request(url: str, method: str, auth_token: str, data: JSONType):
+    valid_url = is_valid_url(url)
+
+    if not valid_url:
+        return "Error: Invalid URL"
+
+    sanitized_url: str = sanitize_url(url)
+
+    print("Starting HTTP request...")
```

Comment:
Please use logger with info level for this. Logging in the application is currently a mess and we have to start improving somewhere. :)